src/userInterface/userInterface.ts

```diff
@@ -205,6 +207,12 @@ export class RosenUserInterface implements IRosenUserInterface {
     const chains = tokensInChain[0];
     const destChainToken = chains[targetChain];
 
+    if (destChainToken === null) {
+      throw new ChainNotSupportedException(
+        "Token is not supported on destination chain"
+      );
+    }
+
```

Comment:
accessing unknown key of the object returns `undefined`, therefore this check is not working. Please proceed as you did in `checkTokenSupported` function (using `TokenMap.getID` and try-catch`)